### PR TITLE
fix(outputs.bigquery): Add scope to bigquery and remove timeout context

### LINKED ISSUE #14957: Bigquery ADC has missing scope and incorrect context
### Relevant telegraf.conf

```toml
[[outputs.bigquery]]
  project = "my-project"
  dataset = "test"
```

### Logs from Telegraf

```text
2024/03/08 02:15:00 E! inserting metric "<Trim>" failed: Post "https://bigquery.googleapis.com/bigquery/v2/projects/<trim>/datasets/test/tables/<trim>/insertAll?alt=json&prettyPrint=false": Post "https://oauth2.googleapis.com/token": context canceled
```

### System info

Telegraf latest on alpine

### Docker

_No response_

### Steps to reproduce

1. Create a bigquery output plugin field in telegraf.conf making sure credentials file is omitted. 
2. Observe the logs, and it will report incorrect response from STS server or contextcanclled.
...

### Expected behavior

- ADC should be able to discover the default credentials and use that to exchange the token for the required scope.

### Actual behavior

Token exchange fails.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by tgupta3**
## Summary
- Scope was missing for bigquery Client causing ADC to fail.
- Timeout context needs to be removed from connection init, since the dialing happens asynchronously.  https://cloud.google.com/go/docs/reference/cloud.google.com/go/0.94.1#hdr-Timeouts_and_Cancellation

## Checklist

- [x] No AI generated code was used in this PR

## Related issues
resolves #14957

### CODE REVIEW

**Comment by tgupta3** [comment]
> Thanks so much for the pull request! 🤝 ✒️ Just a reminder that the CLA has not yet been signed, and we'll need it before merging. Please [sign the CLA](https://www.influxdata.com/legal/cla/) when you get a chance, then post a comment here saying `!signed-cla`

!signed-cla

File: plugins/outputs/bigquery/bigquery.go [hunk 1]
```diff
@@ -89,13 +89,11 @@ func (s *BigQuery) setUpDefaultClient() error {
 	var credentialsOption option.ClientOption
 
 	ctx := context.Background()
-	ctx, cancel := context.WithTimeout(ctx, time.Duration(s.Timeout))
-	defer cancel()
```
Comment (srebhan) [inline plugins/outputs/bigquery/bigquery.go:93]: Why do you remove the timeout?

**Review by srebhan — COMMENTED** [review]
@tgupta3 thanks for your contribution! I do have one question in the code...

File: plugins/outputs/bigquery/bigquery.go (hunk 1 above)
Comment (tgupta3) [reply plugins/outputs/bigquery/bigquery.go:93]: From the cloud [docs](https://cloud.google.com/go/docs/reference/cloud.google.com/go/0.94.1#hdr-Timeouts_and_Cancellation), 
```
Do not attempt to control the initial connection (dialing) of a service by setting a timeout on the context passed to NewClient. Dialing is non-blocking, so timeouts would be ineffective and would only interfere with credential refreshing, which uses the same context.
```

Comment (tgupta3) [reply plugins/outputs/bigquery/bigquery.go:93]: Done, not sure why the checks didn't trigger though.

Comment (srebhan) [reply plugins/outputs/bigquery/bigquery.go:93]: Please run `make fmt`, seems like go has some different view on how things should be formatted. ;-)

Comment (tgupta3) [reply plugins/outputs/bigquery/bigquery.go:93]: Done

**Review by powersj — APPROVED** [review]

**Pull request merged**